odoo/addons/base/models/ir_ui_view.py

```diff
@@ -555,25 +570,33 @@ def get_inheriting_views_arch(self, view_id, model):
                 for view in views.sudo()
                 if not view.groups_id or (view.groups_id & user_groups)]
 
-    @api.model
-    def raise_view_error(self, message, view_id):
-        view = self.browse(view_id)
-        not_avail = _('n/a')
-        message = (
-            "%(msg)s\n\n" +
-            _("Error context:\nView `%(view_name)s`") +
-            "\n[view_id: %(viewid)s, xml_id: %(xmlid)s, "
-            "model: %(model)s, parent_id: %(parent)s]"
-        ) % {
-            'view_name': view.name or not_avail,
-            'viewid': view_id or not_avail,
-            'xmlid': view.xml_id or not_avail,
-            'model': view.model or not_avail,
-            'parent': view.inherit_id.id or not_avail,
-            'msg': message,
-        }
-        _logger.info(message)
-        raise ValueError(message)
+    def raise_view_error(self, message):
+        log_view_error(message, self)
+        actions = []
+        if not self and not self.env.context.get('inherit_branding'):
+            # an error occurred when loading a default view, find out
+            # especifically which one broke
+            args = self.env.context.get('load_view_args')
+            if args:
+                # reload the view to trigger a ViewError, this time with
+                # branding to properly find the view that causes the error
+                args[0].with_context(inherit_branding=True).load_views(*args[1:])
```

Comment:
Does this second call to `load_views` still serve a purpose? It does not seem to do what the comment says, as nothing apparently needs/uses the branding. The various calls to raise_view_error() were already providing the leaf view that triggers the error, no?
It still seems to work and point to the leaf view when I remove the second call, too.

Reply:
Yes, if you look at the places where `raise_view_error` is called, you'll see that in the method `post_process_and_fields` there are two instances where we get the `data-oe-id` attribute from an element in the view corresponding to a non-existing field.

This is because you can load a view without providing a `view_id`, if this is the case and any inherited view references a non-existing field, `load_views` will crash but it won't have any way of knowing *which* inherited view triggered the crash.

Basically, this is not needed for most cases, but it is the case that @dbo-odoo needed covered (see task chatter)